test_project/test_app/tests.py

```diff
@@ -0,0 +1,149 @@
+            username='unauthorized_user'
+        )
+
+        cls.factory = RequestFactory()
+
+    def test_unauthorized_user_does_not_have_permission(self):
+        self.assertFalse(self.unauthorized_user.has_perm(RestrictedModelPermission))
+
+    def test_authorized_user_has_permission(self):
+        self.assertTrue(self.authorized_user.has_perm(RestrictedModelPermission))
+
+    def test_unauthorized_user_accessing_view(self):
+        request = self.factory.get('/restricted_view')
+        request.user = self.unauthorized_user
+        response = restricted_view(request)
+
+        self.assertEqual(response.status_code, 302)
+
+    def test_authorized_user_accessing_view(self):
+        request = self.factory.get('/restricted_view')
+        request.user = self.authorized_user
+        response = restricted_view(request)
+
+        self.assertEqual(response.status_code, 200)
+
+class GroupLevelTests(TestCase):
+
+    def setUp(self):
+        self.authorized_user = get_user_model().objects.create(
+            username='authorized_user'
+        )
+        self.unauthorized_user = get_user_model().objects.create(
+            username='unauthorized_user'
+        )
+        create_permissions.Command().handle()
+        create_groups.Command().handle()
+
+        self.authorized_user.groups.add(AuthorizedGroup.get_or_create()[0].pk)
+
+        self.factory = RequestFactory()
+
+    def test_authorized_user_has_permission(self):
+        self.assertTrue(self.authorized_user.has_perm(RestrictedModelPermission))
+
+    def test_unauthorized_user_does_not_have_permission(self):
+        self.assertFalse(self.unauthorized_user.has_perm(RestrictedModelPermission))
+
+    def test_unauthorized_user_accessing_view(self):
+        request = self.factory.get('/restricted_view')
+        request.user = self.unauthorized_user
+        response = restricted_view(request)
+
+        self.assertEqual(response.status_code, 302)
+
+    def test_authorized_user_accessing_view(self):
+        request = self.factory.get('/restricted_view')
+        request.user = self.authorized_user
+        response = restricted_view(request)
+
+        self.assertEqual(response.status_code, 200)
+
+class GlobalLevelTests(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.authorized_user = get_user_model().objects.create(
+            username='authorized_user'
+        )
+        perm, created_at = GlobalPermission.get_or_create()
+        cls.authorized_user.user_permissions.add(perm)
+
+        cls.unauthorized_user = get_user_model().objects.create(
+            username='unauthorized_user'
+        )
+
+        cls.factory = RequestFactory()
+
+    def test_unauthorized_user_does_not_have_permission(self):
+        self.assertFalse(self.unauthorized_user.has_perm(GlobalPermission))
+
+    def test_authorized_user_has_permission(self):
+        self.assertTrue(self.authorized_user.has_perm(GlobalPermission))
```

Comment:
[This test is currently failing](https://travis-ci.com/serioeseGmbH/serious-django-permissions/builds/98599055). Yet, when checking the permissions of `self.authorized_user`, the global permission shows up:
```
/serious-django-permissions/test_project/test_app/tests.py(136)test_authorized_user_has_permission()
-> self.assertTrue(self.authorized_user.has_perm(GlobalPermission))
(Pdb) Permission.objects.filter(user=self.authorized_user)
<QuerySet [<Permission: auth | permission | Enables the user to write in any model>]>
```
Is this possibly a problem of `.has_perm()` not recognizing the `GlobalPermission` in the right way? @cobalamin